bson/json_util.py

```diff
@@ -415,13 +379,141 @@ def object_hook(dct, json_options=DEFAULT_JSON_OPTIONS):
     return dct
 
 
+def _get_binary(doc, json_options):
+    if isinstance(doc["$type"], int):
+        doc["$type"] = "%02x" % doc["$type"]
+    subtype = int(doc["$type"], 16)
+    if subtype >= 0xffffff80:  # Handle mongoexport values
+        subtype = int(doc["$type"][6:], 16)
+    data = base64.b64decode(doc["$binary"].encode())
+    # special handling for UUID
+    if subtype == OLD_UUID_SUBTYPE:
+        if json_options.uuid_representation == CSHARP_LEGACY:
+            return uuid.UUID(bytes_le=data)
+        if json_options.uuid_representation == JAVA_LEGACY:
+            data = data[7::-1] + data[:7:-1]
+        return uuid.UUID(bytes=data)
+    if subtype == UUID_SUBTYPE:
+        return uuid.UUID(bytes=data)
+    return Binary(data, subtype)
+
+
+def _get_date(doc, json_options):
+    dtm = doc["$date"]
+    # mongoexport 2.6 and newer
+    if isinstance(dtm, string_type):
+        # Parse offset
+        if dtm[-1] == 'Z':
+            dt = dtm[:-1]
+            offset = 'Z'
+        elif dtm[-3] == ':':
+            # (+|-)HH:MM
+            dt = dtm[:-6]
+            offset = dtm[-6:]
+        elif dtm[-5] in ('+', '-'):
+            # (+|-)HHMM
+            dt = dtm[:-5]
+            offset = dtm[-5:]
+        elif dtm[-3] in ('+', '-'):
+            # (+|-)HH
+            dt = dtm[:-3]
+            offset = dtm[-3:]
+        else:
+            dt = dtm
+            offset = ''
+
+        aware = datetime.datetime.strptime(
+            dt, "%Y-%m-%dT%H:%M:%S.%f").replace(tzinfo=utc)
+
+        if offset and offset != 'Z':
+            if len(offset) == 6:
+                hours, minutes = offset[1:].split(':')
+                secs = (int(hours) * 3600 + int(minutes) * 60)
+            elif len(offset) == 5:
+                secs = (int(offset[1:3]) * 3600 + int(offset[3:]) * 60)
+            elif len(offset) == 3:
+                secs = int(offset[1:3]) * 3600
+            if offset[0] == "-":
+                secs *= -1
+            aware = aware - datetime.timedelta(seconds=secs)
+
+        if json_options.tz_aware:
+            if json_options.tzinfo:
+                aware = aware.astimezone(json_options.tzinfo)
+            return aware
+        else:
+            return aware.replace(tzinfo=None)
+    # mongoexport 2.6 and newer, time before the epoch (SERVER-15275)
+    elif isinstance(dtm, collections.Mapping):
+        millis = int(dtm["$numberLong"])
+    # mongoexport before 2.6
+    else:
+        millis = int(dtm)
+    return bson._millis_to_datetime(millis, json_options)
+
+
+def _get_dbpointer(doc, json_options):
+    dbref = doc['$dbPointer']
+    if isinstance(dbref, DBRef):
```

Comment:
Why would the value of $dbPointer be a DBRef instance? Shouldn't it just be a dict like any other JSON object?